pkg/curriculum/curriculum.go

```diff
@@ -48,24 +53,75 @@ func NewRouter(s rest.Server, store Store) *chi.Mux {
 	server := server{s, store}
 	r := chi.NewRouter()
 	r.Method("POST", "/areas", server.createArea())
-	r.Method("PATCH", "/areas/{areaId}", server.patchArea())
-	r.Method("GET", "/areas/{areaId}", server.getArea())
-	r.Method("DELETE", "/areas/{areaId}", server.deleteArea())
-	r.Method("GET", "/areas/{areaId}/subjects", server.getAreaSubjects())
-	r.Method("POST", "/areas/{areaId}/subjects", server.createSubject())
-
-	r.Method("GET", "/subjects/{subjectId}", server.getSubject())
-	r.Method("PUT", "/subjects/{subjectId}", server.replaceSubject())
-	// TODO: This is not used, refactor after curriculum got e2e test.
-	r.Method("PATCH", "/subjects/{subjectId}", server.updateSubject())
-	r.Method("DELETE", "/subjects/{subjectId}", server.deleteSubject())
-	r.Method("GET", "/subjects/{subjectId}/materials", server.getSubjectMaterials())
-	r.Method("POST", "/subjects/{subjectId}/materials", server.createNewMaterial())
-
-	r.Method("PATCH", "/materials/{materialId}", server.updateMaterial())
+	r.Route("/areas/{areaId}", func(r chi.Router) {
+		r.Use(server.areaMiddleware())
+		r.Method("PATCH", "/", server.patchArea())
+		r.Method("GET", "/", server.getArea())
+		r.Method("DELETE", "/", server.deleteArea())
+		r.Method("GET", "/subjects", server.getAreaSubjects())
+		r.Method("POST", "/subjects", server.createSubject())
+	})
+
+
+	r.Route("/subjects/{subjectId}", func(r chi.Router) {
+		r.Use(server.subjectMiddleware())
+		r.Method("GET", "/", server.getSubject())
+		r.Method("PUT", "/", server.replaceSubject())
+		// TODO: This is not used, refactor after curriculum got e2e test.
+		//r.Method("PATCH", "/subjects/{subjectId}", server.updateSubject())
+		r.Method("DELETE", "/", server.deleteSubject())
+		r.Method("GET", "/materials", server.getSubjectMaterials())
+		r.Method("POST", "/materials", server.createNewMaterial())
+
+		r.Method("PATCH", "/materials/{materialId}", server.updateMaterial())
```

Comment:
I don't think we can put this here. This would change the route from `/materials/{materialId}` to `/subjects/{subjectId}/materials/{materialId}`. I think this route probably should have its own middleware.

Reply:
When will the this endpoint used? Now we have feature of updating materials name on the UI, but the endpoint used is not the PATCH material

Reply:
Ah, you're right, commenting that one out is fine.